File: show/src/show/cName.java
package show;
	public class cName {
  	enum nameType { HON,HON2,HON3,FIRST,MIDDLE,MIDDLE2,LAST,SUFFIX,ANY };
  	int nickName;
  	String hon,hon2,hon3;
  	String first;
  	String middle,middle2;
  	String last;
  	String suffix;
  	String any;

 		// identifySpeakers
  	String hn(String namePartName,String namePart,boolean printShort)
  	{
  	  if (namePart.isEmpty()) return "";
 	    return ((printShort) ? "" : (namePartName+":") ) + namePart +" ";
  	}

  	String print(boolean printShort)
  	{
  		String  message="";
  		message+=hn("H1",hon,printShort);
  		message+=hn("H2",hon2,printShort);
  		message+=hn("H3",hon3,printShort);
  		message+=hn("F",first,printShort);
  		message+=hn("M1",middle,printShort);
  		message+=hn("M2",middle2,printShort);
  		message+=hn("L",last,printShort);
  		message+=hn("S",suffix,printShort);
  		message+=hn("A",any,printShort);
  	  if (nickName>=0 && !printShort)
  	    message+="["+nickName+"]";
  	  return message;
  	}

		public cName(LittleEndianDataInputStream rs) {
			nickName = rs.readInteger();
			hon = rs.readString();
			hon2 = rs.readString();
			hon3 = rs.readString();
			first = rs.readString();
			middle = rs.readString();
			middle2 = rs.readString();
			last = rs.readString();
			suffix = rs.readString();
			any = rs.readString();
		}
	}

File: Web/pyLPBackEnd/EclipseWorkspace/pyLPBackEnd/src/CName.py
"""CName.py - Honorific / first / middle / last / suffix name parts.

Overview:
    Mirrors C++ cName. Slot constants HON..ANY identify which part a
    matcher is looking at. print() builds a short or labelled string;
    toJSON() is the web-API shape.

Pipeline position:
    Embedded in CObject when a resolved entity is a person name.
    identifySpeakers (C++ side) is the producer.

Key entry points:
    - hn() - format one non-empty part
    - print() - concatenate all parts
    - __init__(rs) - read nickName + nine strings
    - toJSON() - dict for the LPWeb / pyLP JSON surface
"""
class CName:
    HON=1
    HON2=2
    HON3=3
    FIRST=4
    MIDDLE=5
    MIDDLE2=6
    LAST=7
    SUFFIX=8
    ANY=9
    
    # identifySpeakers
    # Return "" if namePart is empty; else "namePartName:namePart "
    # (label omitted when printShort).
    def hn(self,namePartName,namePart,printShort):
        if len(namePart)==0: return ""
        return ("" if (printShort) else (namePartName+":") ) + namePart +" "

    # Concatenate H1..A parts; append [nickName] unless printShort.
    def print(self,printShort):
        message=""
        message += self.hn("H1",self.hon,printShort)
        message += self.hn("H2",self.hon2,printShort)
        message += self.hn("H3",self.hon3,printShort)
        message += self.hn("F",self.first,printShort)
        message += self.hn("M1",self.middle,printShort)
        message += self.hn("M2",self.middle2,printShort)
        message += self.hn("L",self.last,printShort)
        message += self.hn("S",self.suffix,printShort)
        message += self.hn("A",self.any,printShort)
        if (self.nickName>=0 and not printShort):
            message += "["+str(self.nickName)+"]"
        return message

    # Deserialize nickName (int) plus hon/hon2/hon3/first/middle/middle2/
    # last/suffix/any strings from LPIO `rs`.
    def __init__(self,rs):
        self.nickName = rs.read_integer()
        self.hon = rs.read_string()
        self.hon2 = rs.read_string()
        self.hon3 = rs.read_string()
        self.first = rs.read_string()
        self.middle = rs.read_string()
        self.middle2 = rs.read_string()
        self.last = rs.read_string()
        self.suffix = rs.read_string()
        self.any = rs.read_string()
        
    def write(self,rs):
    def toJSON(self):
        ret = {}
        ret['nickName'] = self.nickName
        ret['hon'] = self.hon
        ret['hon2'] = self.hon2
        ret['hon3'] = self.hon3
        ret['first'] = self.first
        ret['middle'] = self.middle
        ret['middle2'] = self.middle2
        ret['last'] = self.last
        ret['suffix'] = self.suffix
        ret['any'] = self.any
        return ret
